packages/orchestration/test/test-chain-name-hub.js

```diff
@@ -0,0 +1,213 @@
+// @ts-check
+import { test as anyTest } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+import { makeNameHubKit } from '@agoric/vats';
+import { mockChainInfo } from './mocks/chain-info.js';
+
+/**
+ * @import {Brand} from '@agoric/ertp/exported.js';
+ * @import {IBCConnectionID} from '@agoric/vats';
+ */
+
+const test = anyTest;
+
+/** @typedef {`channel-${number}`} IBCChannelID */
+/** @typedef {`ibc/${string}`} IBCDenom */
+
+/**
+ * // TODO import from cosmic-proto?
+ * @typedef {'STATE_OPEN' | 'STATE_TRYOPEN' | 'STATE_INIT' | 'STATE_CLOSED'} IBCConnectionState
+ */
+
+/**
+ * @typedef {{
+ *  counterpartyChannelId: IBCChannelID;
+ *  counterpartyPortId: string;
+ *  sourceChannelId: IBCChannelID;
+ *  sourcePortId: string;
+ *  version: string;
+ *  state: IBCConnectionState;
+ *  ordering: 'ORDER_UNORDERED' | 'ORDER_ORDERED';
+ * }} IBChannelInfo
+ */
+
+/**
+ * @typedef {{
+ *  clientId: string;
+ *  counterparty: {
+ *    clientId: string;
+ *    connectionId: string;
+ *  };
+ *  id: IBCConnectionID;
+ *  state: IBCConnectionState
+ *  revisionHeight: bigint;
+ *  revisionNumber: bigint;
+ * }} IBCConnectionInfo;
+ */
+
+/**
+ * @type {{
+ *  chainId: string;
+ *  bondAsset: {
+ *    brand: Brand;
+ *    denom: IBCDenom;
+ *  };
+ *  brand: {
+ *    [denom: IBCDenom]: Brand;
+ *  };
+ *  denom: {
+ *    [allegedBrandName: string]: IBCDenom;
```

Comment:
alleged names should only be used for debugging, not for anything that affects correctness

Reply:
What is the correct name for `FOO` when we do `E(agoricNames).lookup('brand', 'FOO')`?

Reply:
issuerName, in the case of brands

more generally, just name (or key)